src/Target.cpp

```diff
@@ -788,6 +788,88 @@ void bad_target_string(const std::string &target) {
 
 }  // namespace
 
+void Target::validate_features() const {
+    // Note that the features don't have to be exhaustive, but enough to avoid obvious mistakes is good.
+    if (arch == X86) {
+        static const std::vector<Feature> bad_features_x86 = {
+            ARMDotProd,
+            ARMFp16,
+            ARMv7s,
+            ARMv81a,
+            NoNEON,
+            POWER_ARCH_2_07,
+            RVV,
+            SVE,
+            SVE2,
+            VSX,
+            WasmBulkMemory,
+            WasmSatFloatToInt,
+            WasmSignExt,
+            WasmSimd128,
+            WasmThreads,
+        };
+        user_assert(!features_any_of(bad_features_x86)) << "At least one of the features for "
```

Comment:
A small code change would take the guesswork out of handling this error:
```
for  (Feature f : bad_features_x86) {
   user_assert(!has_feature(f)) << "Target feature " << f ...
}
```

Reply:
Done